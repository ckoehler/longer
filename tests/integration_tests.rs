use assert_cmd::Command;

#[test]
fn test_basic_calculation() {
    let mut cmd = Command::cargo_bin("longer").unwrap();
    let output = cmd
        .arg("--start")
        .arg("1990-01-01")
        .arg("--event")
        .arg("2000-01-01")
        .output()
        .unwrap();

    assert!(output.status.success());
    insta::assert_snapshot!(String::from_utf8_lossy(&output.stdout));
}

#[test]
fn test_different_date_range() {
    let mut cmd = Command::cargo_bin("longer").unwrap();
    let output = cmd
        .arg("--start")
        .arg("1985-03-15")
        .arg("--event")
        .arg("2010-07-22")
        .output()
        .unwrap();

    assert!(output.status.success());
    insta::assert_snapshot!(String::from_utf8_lossy(&output.stdout));
}

#[test]
fn test_invalid_start_date() {
    let mut cmd = Command::cargo_bin("longer").unwrap();
    let output = cmd
        .arg("--start")
        .arg("invalid-date")
        .arg("--event")
        .arg("2000-01-01")
        .output()
        .unwrap();

    assert!(!output.status.success());
    insta::assert_snapshot!(String::from_utf8_lossy(&output.stderr));
}

#[test]
fn test_invalid_event_date() {
    let mut cmd = Command::cargo_bin("longer").unwrap();
    let output = cmd
        .arg("--start")
        .arg("1990-01-01")
        .arg("--event")
        .arg("not-a-date")
        .output()
        .unwrap();

    assert!(!output.status.success());
    insta::assert_snapshot!(String::from_utf8_lossy(&output.stderr));
}

#[test]
fn test_event_before_start() {
    let mut cmd = Command::cargo_bin("longer").unwrap();
    let output = cmd
        .arg("--start")
        .arg("2000-01-01")
        .arg("--event")
        .arg("1990-01-01")
        .output()
        .unwrap();

    assert!(!output.status.success());
    insta::assert_snapshot!(String::from_utf8_lossy(&output.stderr));
}

#[test]
fn test_help_output() {
    let mut cmd = Command::cargo_bin("longer").unwrap();
    let output = cmd.arg("--help").output().unwrap();

    assert!(output.status.success());
    insta::assert_snapshot!(String::from_utf8_lossy(&output.stdout));
}
