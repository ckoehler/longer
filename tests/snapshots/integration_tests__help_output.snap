---
source: tests/integration_tests.rs
expression: "String::from_utf8_lossy(&output.stdout)"
---
Have you ever wondered when you will have been married longer than not? This tool calculates that date for you!

Usage: longer --start <YYYY-MM-DD> --event <YYYY-MM-DD>

Options:
  -s, --start <YYYY-MM-DD>
          The start date, e.g. your birthday
          
          Format: YYYY-MM-DD (e.g., 1990-05-15)

  -e, --event <YYYY-MM-DD>
          The event to count as the midpoint, e.g. your wedding date
          
          Format: YYYY-MM-DD (e.g., 2015-08-20)

  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version

EXAMPLES:
  longer --start 1990-05-15 --event 2015-08-20
  longer -s 1985-01-01 -e 2010-06-15

DATE FORMAT:
  All dates must be in YYYY-MM-DD format (ISO 8601)
